sql-plugin/src/main/scala/org/apache/spark/sql/rapids/AggregateFunctions.scala

```diff
@@ -569,6 +580,63 @@ case class GpuMax(child: Expression) extends GpuAggregateFunction
   }
 }
 
+/** Max aggregation for non FloatType or DoubleType */
+case class GpuBasicMax(child: Expression) extends GpuMax(child)
+
+/** Max aggregation for FloatType and DoubleType */
+case class GpuFloatMax(child: Expression) extends GpuMax(child) 
+    with GpuReplaceWindowFunction{
+  override lazy val inputProjection: Seq[Expression] = child.dataType match {
+    case FloatType | DoubleType => Seq(child, GpuIsNan(child))
+    case _ => Seq(child)
+  }
+
+  protected lazy val updateIsNan = new CudfMax(BooleanType)
+  protected lazy val updateMaxVal = new CudfMax(dataType)
+  override lazy val updateAggregates: Seq[CudfAggregate] = child.dataType match {
+    case FloatType | DoubleType => Seq(updateMaxVal, updateIsNan)
+    case _ => Seq(updateMaxVal)
+  }
+
+  override lazy val postUpdate: Seq[Expression] = child.dataType match {
+    case FloatType => Seq(GpuIf(updateIsNan.attr, GpuLiteral(Float.NaN, FloatType), updateMaxVal.attr))
+    case DoubleType => Seq(GpuIf(updateIsNan.attr, GpuLiteral(Double.NaN, DoubleType), updateMaxVal.attr))
+    case _ => postUpdateAttr
+  }
+
+  override lazy val preMerge: Seq[Expression] = child.dataType match {
+    case FloatType | DoubleType => Seq(evaluateExpression, GpuIsNan(evaluateExpression))
+    case _ => aggBufferAttributes
+  } 
+  protected lazy val mergeIsNan = new CudfMax(BooleanType)
```

Comment:
nit: Could we create a CudfAny that really is just a CudfMax(BooleanType). This is just for readability of the code.

```
object CudfAny {
  def apply(): CudfAggregation = new CudfMax(BooleanType)
}
```

Reply:
Added!